yb-voyager/cmd/exportSchema.go

```diff
@@ -195,24 +201,75 @@ func exportSchema() error {
 	return nil
 }
 
-func nudgeUserToRunAssessment() error {
-	msr, err := metaDB.GetMigrationStatusRecord()
-	if err != nil {
-		return fmt.Errorf("failed to get migration status record: %w", err)
+func runAssessMigrationCmdBeforExportSchemaIfRequired(exportSchemaCmd *cobra.Command) error {
+	if !bool(assessSchemaBeforeExport) {
+		log.Infof("skipping running assess-migration command before export schema as flag --assess-schema-before-export is set as false.")
+		return nil
 	}
 
-	if msr.MigrationAssessmentDone {
+	if source.DBType != POSTGRESQL {
+		log.Infof("skipping running assess-migration command before export schema as source DB type is not PostgreSQL.")
 		return nil
 	}
-	if source.DBType != POSTGRESQL {
-		// At this point, we are only interested in nudging postgres users to run assessment
-		// because it is far more advanced and useful for them.
+
+	if ok, _ := IsMigrationAssessmentDoneDirectly(metaDB); ok {
+		log.Infof("migration assessment is already done, skipping running assess-migration command.")
+		return nil
+	} else if ok, _ := IsMigrationAssessmentDoneViaExportSchema(); ok {
+		log.Infof("migration assessment is already done via export schema, skipping running assess-migration command.")
 		return nil
 	}
-	if !utils.AskPrompt("It is recommended to run assess-migration before exporting schema. https://docs.yugabyte.com/preview/yugabyte-voyager/migrate/assess-migration/ \n" +
-		"Do you want to continue anyway without running assess-migration") {
-		utils.ErrExit("Aborting...")
+
+	var assessFlagsWithValues []string
+	commonFlags := utils.GetCommonFlags(exportSchemaCmd, assessMigrationCmd)
+	for _, flag := range commonFlags {
+		// don't pass start-clean flag to assess-migration command here
+		if flag.Name == "start-clean" || !flag.Changed {
+			continue
+		}
+
+		// bool flags: --flag=value
+		// everything else: --flag value
+		if flag.Value.Type() == "bool" {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				fmt.Sprintf("--%s=%s", flag.Name, flag.Value.String()),
+			)
+		} else {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				"--"+flag.Name,
+				flag.Value.String(),
+			)
+		}
 	}
+
+	// Append --yes=true(irrespective) at the end to override any --yes=false if set in export schema cmd
+	assessFlagsWithValues = append(assessFlagsWithValues,
+		"--assessment-invoked-from-export-schema", "true",
+		"--iops-capture-interval", "0", // TODO: any small but significant duration will be better than 0
+		"--target-db-version", ybversion.LatestStable.String(),
+		"--yes",
+	)
+
+	// locate voyager binary
+	voyagerExecutable, err := os.Executable()
+	if err != nil {
+		log.Warnf("cannot locate executable, skipping assessment: %v", err)
```

Comment:
let's return the error here properly?  From the perspective of this function, it should return the error.
The caller function is responsible for deciding how to handle that error (ignore, or raise it further).